gravitee-apim-console-webui/src/management/home/home-api-health-check/home-api-health-check.component.ts

```diff
@@ -240,41 +256,81 @@ export class HomeApiHealthCheckComponent implements OnInit, OnDestroy {
     this.filters$.next(this.filters);
   }
 
-  private toApisTableDS(api: PagedResult<Api>): ApisTableDS[] {
-    return api.data.map(
-      (api) =>
-        ({
-          id: api.id,
-          name: api.name,
-          version: api.version,
-          contextPath: api.context_path,
-          tags: api.tags.join(', '),
-          owner: api?.owner?.displayName,
-          ownerEmail: api?.owner?.email,
-          picture: api.picture_url,
-          state: api.state,
-          lifecycleState: api.lifecycle_state,
-          workflowBadge: this.getWorkflowBadge(api),
-          healthcheck_enabled: api.healthcheck_enabled,
-          origin: api.definition_context.origin,
-          availability$: this.getAvailability$(api),
-        }) as ApisTableDS,
-    );
+  private toApisTableDS(apiR: ApisResponse): ApisTableDS[] {
+    return apiR.data.flatMap((api) => {
+      switch (api.definitionVersion) {
+        case 'FEDERATED':
+          return [];
+        case 'V1':
+          return [];
+        case 'V2':
+          return [this.v2toApisTableDS(api)];
+        case 'V4':
+          return this.v4toApisTableDS(api);
+      }
+    });
   }
 
-  private getWorkflowBadge(api) {
-    const state = api.lifecycle_state === 'DEPRECATED' ? api.lifecycle_state : api.workflow_state;
+  private v2toApisTableDS(api: ApiV2): ApisTableDS {
+    return {
+      id: api.id,
+      name: api.name,
+      definitionVersion: api.definitionVersion,
+      version: api.apiVersion,
+      tags: api.tags.join(', '),
+      owner: api?.primaryOwner.displayName,
+      ownerEmail: api?.primaryOwner.email,
+      state: api.state,
+      lifecycleState: api.lifecycleState,
+      workflowBadge: this.getWorkflowBadge(api),
+      availability$: this.getAvailability$(api.id, this.healthcheckEnabledApiV2(api)),
+      picture: api._links.pictureUrl,
+      healthcheck_enabled: this.healthcheckEnabledApiV2(api),
+      origin: api.originContext.origin,
+    } satisfies ApisTableDS;
+  }
+
+  private v4toApisTableDS(api: ApiV4): ApisTableDS[] {
```

Comment:
Is there a particular reason to return an array here and not in `v2toApisTableDS`